package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -25,22 +25,23 @@ if dns.servers then
 	})
 end
 
-if next_node.name and next_node.ip4 then
-	uci:section('dhcp', 'domain', 'nextnode4', {
-		name = next_node.name,
-		ip = next_node.ip4,
+local function set_dns_record(name, ip, sectionname)
+	if not ip then return end
+	uci:section('dhcp', 'domain', sectionname, {
+		name = name,
+		ip = ip,
 	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode4')
 end
 
-if next_node.name and next_node.ip6 then
-	uci:section('dhcp', 'domain', 'nextnode6', {
-		name = next_node.name,
-		ip = next_node.ip6,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode6')
+uci:foreach("dhcp", "domain", function(s)
+	if (s['.name'] and string.match(s['.name'], "^nextnode[46]_")) then
```

Comment:
Remove the underscore from the match, so this will also remove old "nextnode4"/"nextnode6" sections.

Reply:
done.